### PR TITLE
Fix `errorOverlayText` prop on small screens

### PR DESCRIPTION
**Pull request by dleroux**
### WHY are these changes introduced?

Because we render different components to output `overlayText` and `errorOverlayText` depending on screen size, the `errorOverlayText` rendered the wrong text.

### WHAT is this pull request doing?

Ensures the right text is rendered and tests are added to prevent this in the future.

## 

### How to 🎩
<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import * as React from 'react';
import {DropZone, Stack, Thumbnail, Caption} from '@shopify/polaris';

interface State {
  dirty: boolean;
  field: string;
  saving: boolean;
}

export default class DropZoneExample extends React.Component {
  state = {
    files: [],
  };

  render() {
    const {files} = this.state;
    const validImageTypes = ['image/gif', 'image/jpeg', 'image/png'];

    const fileUpload = !files.length && <DropZone.FileUpload />;
    const uploadedFiles = files.length > 0 && (
      <Stack vertical>
        {files.map((file, index) => (
          <Stack alignment="center" key={index}>
            <Thumbnail
              size="small"
              alt={file.name}
              source={
                validImageTypes.indexOf(file.type) > 0
                  ? window.URL.createObjectURL(file)
                  : 'https://cdn.shopify.com/s/files/1/0757/9955/files/New_Post.png?12678548500147524304'
              }
            />
            <div>
              {file.name} <Caption>{file.size} bytes</Caption>
            </div>
          </Stack>
        ))}
      </Stack>
    );

    return (
      <DropZone
        accept="image/*"
        type="image"
        onDrop={(files, acceptedFiles, rejectedFiles) => {
          this.setState({files: [...this.state.files, ...acceptedFiles]});
        }}
        errorOverlayText="Custom errorOverlayText"
      >
        {uploadedFiles}
        {fileUpload}
      </DropZone>
    );
  }
}
```

</details>

### 🎩 checklist

* [ ] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
* [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
* [ ] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)
* [x] Updated [UNRELEASED.md](https://github.com/Shopify/polaris-react/blob/master/UNRELEASED.md)

### CODE REVIEW

**Comment by dleroux** [comment]
Just changed to wip to try something different with jest

**Comment by danrosenthal** [comment]
@dleroux test failures look legitimate on this one

**Comment by dleroux** [comment]
Sorry about that, I hadn't noticed the failing CI. Fixed now @danrosenthal

File: src/components/DropZone/tests/DropZone.test.tsx [hunk 1]
```diff
@@ -207,4 +243,94 @@ describe('<DropZone />', () => {
       );
     });
   });
+
+  describe('overlayText', () => {
+    const overlayText = 'overlay text';
+    it("doesn't renders the overlayText on small screens", () => {
```
Comment (solonaarmstrong) [inline src/components/DropZone/tests/DropZone.test.tsx]:
```suggestion
    it('does not render the overlayText on small screens', () => {
```

File: src/components/DropZone/tests/DropZone.test.tsx [hunk 2]
```diff
@@ -207,4 +243,94 @@ describe('<DropZone />', () => {
...
+        <DropZone overlayText={overlayText} />,
+      );
+      triggerDragEnter(dropZone);
+      const displayText = dropZone.find(DisplayText);
+      expect(displayText.contains(overlayText)).toBe(true);
+    });
+  });
+
+  describe('errorOverlayText ', () => {
+    const errorOverlayText = "can't drop this";
```
[the remark this answers is not in the document]
Comment (dleroux) [reply src/components/DropZone/tests/DropZone.test.tsx]: Should have put big pants or music notes here.

**Pull request merged**